File: solutions/cpp/643/solution.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double sum = 0;
        for (int i = 0; i < k; i++) {
            sum += nums[i];
        }

        double average = sum / k;
        for (int i = k; i < nums.size(); i++) {
            sum -= nums[i - k];
            sum += nums[i];
            average = max(average, sum / k);
        }
        return average;
    }
};

File: solutions/golang/643/solution.go
func findMaxAverage(nums []int, k int) float64 {
	sum := 0.0
	for i := 0; i < k; i++ {
		sum += float64(nums[i])
	}

	average := sum / float64(k)
	for i := k; i < len(nums); i++ {
		sum -= float64(nums[i-k])
		sum += float64(nums[i])
		average = max(average, sum/float64(k))
	}
	return average
}